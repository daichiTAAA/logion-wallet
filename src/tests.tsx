import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
