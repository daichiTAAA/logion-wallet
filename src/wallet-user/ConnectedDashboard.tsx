import React from 'react';

import Tokenization from "./Tokenization";
import RefreshRequestsButton from './RefreshRequestsButton';
import PendingTokenizationRequests from './PendingTokenizationRequests';
import AcceptedTokenizationRequests from './AcceptedTokenizationRequests';
import RejectedTokenizationRequests from './RejectedTokenizationRequests';

export default function ConnectedDashboard() {
    return (
        <>
            <Tokenization/>
            <RefreshRequestsButton/>
            <PendingTokenizationRequests />
            <AcceptedTokenizationRequests />
            <RejectedTokenizationRequests />
        </>
    );
}
