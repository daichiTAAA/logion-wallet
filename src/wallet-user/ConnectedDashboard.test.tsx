import { shallowRender } from '../tests';
import ConnectedDashboard from './ConnectedDashboard';

test("renders", () => {
    const tree = shallowRender(<ConnectedDashboard />);
    expect(tree).toMatchSnapshot();
});
