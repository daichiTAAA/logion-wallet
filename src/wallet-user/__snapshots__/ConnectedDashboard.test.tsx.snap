// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<React.Fragment>
  <Tokenization />
  <RefreshRequestsButton />
  <PendingTokenizationRequests />
  <AcceptedTokenizationRequests />
  <RejectedTokenizationRequests />
</React.Fragment>
`;
